server.js

```diff
@@ -14,118 +18,227 @@ app.use(express.json());
+app.post(`${API_BASE_PATH}/disbursement/user`, async (req, res) => {
   try {
     console.log("Received get user request", req.body);
-
     const { phone } = req.body;
-    const data = {
-      return_code: 1,
-      return_message: "Success",
-      sub_return_code: 1,
-      sub_return_message: "Success",
-      data: {
-        name: "User test",
-        phone: phone,
-        m_u_id: uuid(),
-      }
-    };
-    res.json(data); // sending a tuple with orderRef as well to inform about the unique order reference
+    const validatePhoneResult = validatePhone(phone);
+    if (validatePhoneResult !== "") {
+      return res.status(400).json(validatePhoneResult);
+    }
+    if (isMock) {
+      const data = {
+        return_code: 1,
+        return_message: "Success",
+        sub_return_code: 1,
+        sub_return_message: "Success",
+        data: {
+          name: "Longhb",
+          phone: phone,
+          m_u_id: uuid(),
+        }
+      };
+      return res.json(data);
+    }
+    const time = Date.now();
+    const message = [appId, phone, time].join("|");
+    const mac = crypto.HmacSHA256(message, process.env.REACT_APP_KEY1).toString();
+
+    const response = await fetch(`${SANDBOX_URL}/v2/disbursement/user`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ app_id: appId, phone, time, mac })
+    });
+    return res.json(await response.json());
   } catch (err) {
     console.error(`Error: ${err.message}, error code: ${err.errorCode}`);
     res.status(err.statusCode).json(err.message);
   }
 });
 
 // Get merchant balance
-app.post("/v2/disbursement/balance", async (req, res) => {
+app.post(`${API_BASE_PATH}/disbursement/balance`, async (req, res) => {
   try {
     console.log("Received get balance request", req.body);
 
-    const { appId, paymentId } = req.body;
-    const data = {
-      return_code: 1,
-      return_message: "Success",
-      sub_return_code: 1,
-      sub_return_message: "Success",
-      data: {
-        balance: 100000,
-      }
-    };
-    res.json(data); // sending a tuple with orderRef as well to inform about the unique order reference
+    if (isMock) {
+      const data = {
+        return_code: 1,
+        return_message: "Success",
+        sub_return_code: 1,
+        sub_return_message: "Success",
+        data: {
+          balance: 100000,
+        }
+      };
+      return res.json(data);
+    }
+
+    const time = Date.now();
+    const message = [appId, paymentId, time].join("|");
+    const mac = crypto.HmacSHA256(message, process.env.REACT_APP_KEY1).toString();
+    const response = await fetch(`${SANDBOX_URL}/v2/disbursement/balance`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ app_id: appId, payment_id: paymentId, time, mac })
+    });
+    return res.json(await response.json());
```

Comment:
@leonthaibao 
The response.json() method returns a Promise that resolves to a JavaScript object representing the parsed JSON data from the response body. By using the await keyword before calling this method, we tell the JavaScript engine to pause execution of the current function until the Promise is resolved, and then assign the resulting value to a variable.

Without using await, the code would continue executing immediately after calling response.json(), before the Promise has resolved, which could result in the function attempting to use the parsed data before it is available. This could lead to errors or unexpected behavior in the program.